ext/sockets/sockets.c

```diff
@@ -1853,6 +1858,31 @@ PHP_FUNCTION(socket_get_option)
 		}
 	}
 
+#ifdef SOL_FILTER
+	if (level == SOL_FILTER) {
+		switch (optname) {
+
+			case FIL_LIST: {
+				size_t i;
+				struct fil_info fi[32] = {{0}};
+				optlen = sizeof(fi);
+
+				if (getsockopt(php_sock->bsd_socket, level, optname, (char*)fi, &optlen) != 0) {
+					PHP_SOCKET_ERROR(php_sock, "Unable to retrieve socket option", errno);
+					RETURN_FALSE;
+				}
+
+				array_init(return_value);
+
+				for (i = 0; i < optlen / sizeof(struct fil_info); i++)
+					add_index_string(return_value, i, fi[i].fi_name);
```

Comment:
nit: please always use braces:
```suggestion
				for (i = 0; i < optlen / sizeof(struct fil_info); i++) {
					add_index_string(return_value, i, fi[i].fi_name);
				}
```